optuna/study/study.py

```diff
@@ -384,17 +381,25 @@ def objective(trial):
                 Flag to show progress bars or not. To disable progress bar, set this ``False``.
                 Currently, progress bar is experimental feature and disabled
                 when ``n_jobs`` :math:`\\ne 1`.
+            scheduler:
+                Scheduler to determine how trials are executed in parallel when ``n_jobs`` is
+                larger than one. Can be either ``threads`` or ``processes`` for multithreading or
+                multiprocess, respectively. Ignored if ``n_jobs`` is one.
+
+                .. note::
+                    Added in v2.9.0 as an experimental feature. The interface may change in newer
+                    versions without prior notice. See
+                    https://github.com/optuna/optuna/releases/tag/v2.9.0.
```

Comment:
I am wondering if Optuna should provide such a feature.

I think that this feature is too much just for multiprocessing support. If you are planning to support another schedulers (e.g. a scheduler based on dask-distributed and/or Kubernetes jobs), I think we can use Joblib instead as in previous versions.

Reply:
One alternative approach is removing this option and guides users to use `ProcessPoolExecutor` in a documentation and a release note (like https://github.com/optuna/optuna/releases/tag/v2.7.0).

```python
storage = storages.MultiprocessInMemoryStorage()
study = create_study(storage=storage)

with ProcessPoolExecutor(max_workers=5) as executor:
    for _ in range(5):
        executor.submit(study.optimize, objective, 100)
```